tests/integration/test_measurement_integration.py

```diff
@@ -114,3 +114,25 @@ def test_measure_fock(self, setup_eng, cutoff, batch_size):
                 assert np.all(photons_out == np.tile(total_photons, batch_size))
             else:
                 assert np.all(photons_out == total_photons)
+
+    @pytest.mark.backends("gaussian")
+    @pytest.mark.fixture('pure')
+    def test_embed_graph(self, setup_eng, cutoff, batch_size):
+        """Test that a graph embeded with the right photon number gives the right statistics. """
+        n_samples = 10000
+        eng, prog = setup_eng(2)
+        A = np.array([[0.0, 1.0], [1.0, 0.0]])
+        n_mean_per_mode = 1
+        with prog.context as q:
+            ops.GraphEmbed(A, n_mean_per_mode) | (q[0], q[1])
+            ops.MeasureFock() | (q[0], q[1])
```

Comment:
If the goal here is just to test if `GraphEmbed` has the correct mean photon numbers, can we not look at the mean photon numbers directly? e.g., 
```
state = eng.run(prog).state
cov = state.cov()
mean_photon_numbers = some_function(state)
```

Reply:
yes. That should work much better since it is not stochastic.

Reply:
The latest commit should do exactly what you requested.